File: projectEuler/code/0050.py
#!/usr/bin/env python3

#  The prime 41, can be written as the sum of six consecutive primes:
#  41 = 2 + 3 + 5 + 7 + 11 + 13
#  This is the longest sum of consecutive primes that adds to a prime below one-hundred.
#  The longest sum of consecutive primes below one-thousand that adds to a prime, contains 21 terms, and is equal to 953.
#  Which prime, below one-million, can be written as the sum of the most consecutive primes?

import math as m

# check prime with integer
def check_prime(nn):
  n = int(nn)
  stat = 1
  if (n % 2 == 0):
    stat -= 1
    return stat
  for i in range(3, m.ceil(m.sqrt(n))):
    if (n % i == 0):
      stat -= 1
      return stat
  return stat

for i in range(5, 200000, 2):
  rec = 1
  sp = 2
  ind = 1
  for j in range(3, i+1, 2):
    if check_prime(j):
      sp += i
      ind += 1
  if check_prime(sp):
    if sp > rec and sp != 2: 
      rec = sp
      print("new record " + str(rec) + " with lengths " + str(ind))

File: projectEuler/code/0050.rb
#!/usr/bin/env ruby

def isPrime(n)
   if n%2 == 0
      return false
   else
      (3..n-1).step(2).each do |d|
         if n%d == 0
            return false
         end
      end
      return true
   end
end

p_array = [2]

range = (3..10000).step(2)
range.each do |d|
   if isPrime(d)
      p_array << d
   end
end

# sum_record = 2
num_record = 2
l = p_array.length()

n_cons_test = (1..1000)
n_cons_test.each do |nc|
   (0..(l-1-nc)).each do |ncc|
      test_array = p_array[ncc, nc]
      sum = test_array.inject(0, :+)
      if sum > 1000000
         break
      end
      if nc > num_record && isPrime(sum)
         # sum_record = sum
         num_record = nc
         puts "Found sum: " + sum.to_s + " with " + nc.to_s + " primes"
      end
   end
end

